libstb/secvar/storage/secboot_tpm.c

```diff
@@ -363,15 +374,68 @@ static int secboot_tpm_load_bank(struct list_head *bank, int section)
 }
 
 
+/* Ensure the NV indices were defined with the correct set of attributes */
+static int secboot_tpm_check_tpmnv_attrs(void)
+{
+	TPMS_NV_PUBLIC nv_public; /* Throwaway, we only want the name field */
+	TPM2B_NAME nv_vars_name;
+	TPM2B_NAME nv_control_name;
+	int rc;
+
+	rc = tss_nv_read_public(SECBOOT_TPMNV_VARS_INDEX, &nv_public, &nv_vars_name);
+	if (rc) {
+		prlog(PR_ERR, "Failed to readpublic from the VARS index, rc=%d\n", rc);
+		return rc;
+	}
+	rc = tss_nv_read_public(SECBOOT_TPMNV_CONTROL_INDEX, &nv_public, &nv_control_name);
+	if (rc) {
+		prlog(PR_ERR, "Failed to readpublic from the CONTROL index, rc=%d\n", rc);
+		return rc;
+	}
+
+	if (memcmp(tpmnv_vars_name, &nv_vars_name, sizeof(tpmnv_vars_name))) {
+		prlog(PR_ERR, "VARS index not defined with the correct attributes\n");
+		return -1;
```

Comment:
Why -1, shouldn't be some OPAL error ?

Reply:
yes, probably should be `OPAL_INTERNAL_ERROR`